src/main/webapp/teacher/controlGameInstance/controlGameInstance.js

```diff
@@ -188,22 +203,133 @@ async function updateCurrentQuestion({ gameId, currentQuestionId, isCurrentQuest
+    // This helps remove unnecesary listeners to an answers' collection
+    unsubscribeCurrentActiveQuestionAnswersInGameInstance();
+  }
+  unsubscribeCurrentActiveQuestionAnswersInGameInstance = db.collection('gameInstance').doc(gameInstanceId).collection('questions').doc(currentQuestionId).collection('answers')
+  .onSnapshot(function(querySnapshot) {
+    // Clean the answers' div
+    clearCurrentQuestionAnswersDiv();
+
+    querySnapshot.forEach(function(doc) {
+      // Update the current question's answers in the "Active" top panel
+      updateCurrentQuestionAnswerStats(doc.data(), doc.id);
+
+      // Update the current question's answers in the history section
+      updateCurrentQuestionAnswerStatsHistory(doc.data(), doc.id, currentQuestionId);
+    });
+  });
+}
 
+// Clear the current question's answers in UI
+function clearCurrentQuestionAnswersDiv() {
+  const currentQuestionAnswersDivElement = document.getElementById('jsCurrentQuestionAnswers');
+  currentQuestionAnswersDivElement.innerText = '';
+}
+
+// Update the current question's answer's stats in the main panel
+function updateCurrentQuestionAnswerStats(updatedAnswer) {
+  const currentQuestionAnswersDivElement = document.getElementById('jsCurrentQuestionAnswers');
+  
+  const answerElement = document.createElement('div');
+  answerElement.innerText = updatedAnswer.title + ' with ' + updatedAnswer.numberAnswers + ' answers.';
+  
+  if (updatedAnswer.correct) {
+    answerElement.innerText += ' (Correct answer)';
+  }
+
+  currentQuestionAnswersDivElement.appendChild(answerElement);
+}
+
+// Update a question's stats in the history section
+function updateCurrentQuestionAnswerStatsHistory({ updatedAnswer, answerId, questionId } = {}) {
+  const answerInQuestionStatsDivElement = document.getElementById('stats-' + questionId + '-' + answerId);
+  if (answerInQuestionStatsDivElement == null) {
+    // This may be executed before the node is created the first time
+    return;
+  }
+
+  // Update question stats
+  answerInQuestionStatsDivElement.innerText = updatedAnswer.title + ' with ' + updatedAnswer.numberAnswers + ' answers.';
+
+  if (updatedAnswer.correct) {
+    answerInQuestionStatsDivElement.innerText += ' (Correct answer)';
+  } else {
+    answerInQuestionStatsDivElement.innerText += ' (Wrong answer)';
+  }
+}
+
+// Update the number of students in game in UI
 function updateNumberOfMembersUI(numberOfMembers) {
   const numberOfMembersElement = document.getElementById("jsNumberOfStudents");
   numberOfMembersElement.innerText = "There are " + numberOfMembers + " students registered in your room.";
 }
 
+// Listen to the current question's stats for any change
+function initQuestionStatsListener({ gameInstanceId, currentQuestionId } = {}) {
+  if (unsubscribeCurrentActiveQuestionInGameInstance != null) {
+    // This is to stop listening to live changes to the previous question which is not active anymore
+    unsubscribeCurrentActiveQuestionInGameInstance();
```

Comment:
What does calling this do? Is it a function?

Reply:
Yep this is to stop listening to the DocRef of the previous question (supposing you clicked next question), it wouldn't make any difference if we didn't unsubscribe because once we're past a question it can't be updated anymore, but I thought it would still be good to unsubscribe to save some space in memory maybe?